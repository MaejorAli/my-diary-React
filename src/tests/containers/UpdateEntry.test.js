import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import UpdateEntryComponent, { UpdateEntry } from '../../containers/UpdateEntry';
import Root from '../../root';

const props = {
  signupUser: jest.fn(),
  signin: jest.fn(),
  publishEntry: jest.fn(),
  error: false,
  history: { push: jest.fn() },
  status: {
    success: true,
  },
  publishedEnty: '',
  match: {
    params: { id: 1 },
  },
  getAnEntry: jest.fn(),
  updateEntry: jest.fn(),
};

const propsTrue = {
  signupUser: jest.fn(),
  signin: jest.fn(),
  publishEntry: jest.fn(),
  error: true,
  history: { push: jest.fn() },
  status: {
    success: true,
  },
  publishedEnty: '',
  match: {
    params: { id: 1 },
  },
  getAnEntry: jest.fn(),
  updateEntry: jest.fn(),
};
const wrapper = mount(
  <Root>
    <MemoryRouter initialEntries={[{ key: 'test' }]}>
      <UpdateEntryComponent {...propsTrue} />
    </MemoryRouter>
  </Root>,
);

const wrapped = shallow(<UpdateEntry {...props} />);


describe('Update Entry container', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('', () => {
    beforeEach(() => {
      wrapped.find('input').simulate('change', { target: { value: 'gu' } });
      wrapped.find('textarea').first().simulate('change', { target: { value: 'foo' } });
    });

    it('', () => {
      expect(wrapped.find('input').first().prop('value')).toEqual('gu');
      expect(wrapped.find('textarea').first().prop('value')).toEqual('foo');
    });
  });
  describe('simulate handle event', () => {
    it('should simulate submit', () => {
      const submitButton = wrapped.find('#publishEntry');

      const mockedEvent = {
        preventDefault: jest.fn(),
      };

      submitButton.simulate('click', mockedEvent);
    });
  });
});
