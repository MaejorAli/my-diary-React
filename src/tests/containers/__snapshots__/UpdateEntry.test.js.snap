// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Entry container render features view should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "test",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "test",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "test",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(UpdateEntry)
          error={true}
          getAnEntry={[MockFunction]}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          publishEntry={[MockFunction]}
          publishedEnty=""
          signin={[MockFunction]}
          signupUser={[MockFunction]}
          status={
            Object {
              "success": true,
            }
          }
          updateEntry={[MockFunction]}
        >
          <UpdateEntry
            entry={Object {}}
            entryLoading={true}
            error={true}
            getAnEntry={[Function]}
            history={
              Object {
                "push": [MockFunction],
              }
            }
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
            publishEntry={[MockFunction]}
            publishedEnty=""
            signin={[MockFunction]}
            signupUser={[MockFunction]}
            status={
              Object {
                "error": false,
                "success": false,
              }
            }
            updateEntry={[Function]}
            updatedEntry={Object {}}
          >
            <div
              className="error"
            >
              <span>
                <Component />
              </span>
            </div>
          </UpdateEntry>
        </Connect(UpdateEntry)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
